OpenRA.Game/Primitives/RingBuffer.cs

```diff
@@ -0,0 +1,144 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Primitives
+{
+	/// <summary>Fixed size rorating buffer backed by an array.</summary>
+	public class RingBuffer<T> : ICollection<T>, IEnumerable<T>
+	{
+		readonly IComparer<T> comparer;
+		readonly T[] values;
+		int start;
+		int count;
```

Comment:
The names `start` and `count` are weird and not very descriptive. Could they be changed / or explained? Like Start renamed to `CursorPosition` or smth like that?

Reply:
) they are the start index of the first element, and the number of elements in the buffer.
head is worse. cursor position is not what it is.